File: programming/intro.js
// start: declaration
let sqrt = Math.sqrt // ignore
let square = Math.pow // ignore
class Point2D {
  constructor(x, y){
    this.x = x;
    this.y = y
  }
  norm() {
    return sqrt(square(this.x) + square(this.y));
  }
}
// ---sep--- initialization
let pt_a = new Point2D(3, 4);
// ---sep--- extraction
let m = pt_a.norm;
// ---sep--- application
m.call(pt_a)
// result: 5

File: programming/intro.py
# start: declaration
import math # ignore
sqrt = math.sqrt # ignore
def square(x): # ignore
  return x * x # ignore
class Point2D(object):
  def __init__(self, x, y):
    self.x = x
    self.y = y

  def norm(self):
    return sqrt(square(self.x) * square(self.x))
# ---sep--- initialization
pt_a = Point2D(3, 4)
# ---sep--- extraction
m = pt_a.norm
# ---sep--- application
m()
# result: 5
